numpy/_core/tests/test_stringdtype.py

```diff
@@ -842,7 +863,41 @@ def test_datetime_cast(dtype):
         sa = np.delete(sa, 3)
         a = np.delete(a, 3)
 
-    assert_array_equal(sa, a.astype("U"))
+    if input_dtype.startswith("M"):
+        assert_array_equal(sa, a.astype("U"))
+    else:
+        # The timedelta to unicode cast produces strings
+        # that aren't round-trippable and we don't want to
+        # reproduce that behavior in stringdtype
+        assert_array_equal(sa, a.astype("int64").astype("U"))
+
+
+def test_nat_casts():
+    s = 'nat'
+    all_nats = itertools.product(*zip(s.upper(), s.lower()))
+    all_nats = list(map(''.join, all_nats))
+    NaT_dt = np.datetime64('NaT')
+    NaT_td = np.timedelta64('NaT')
+    for na_object in [np._NoValue, None, np.nan, 'nat', '']:
+        # numpy treats empty string and all case combinations of 'nat' as NaT
+        dtype = StringDType(na_object=na_object)
+        arr = np.array([''] + all_nats, dtype=dtype)
+        dt_array = arr.astype('M8[s]')
+        td_array = arr.astype('m8[s]')
+        assert_array_equal(
+            dt_array, np.array([NaT_dt] * arr.size, dtype='M8[s]'))
```

Comment:
In principle, `assert_array_equal` can compare to a scalar, so `assert_array_equal(dt_array, NaT_dt)` should "just work".